config/redis.yml

```diff
@@ -1,9 +1,9 @@
 development:
-  host: localhost
-  port: 6379
+  host: <%= ENV['REDIS_HOST'] || 'localhost' %>
+  port: <%= ENV['REDIS_PORT'] || 6379 %>
 test:
-  host: localhost
-  port: 6379
+  host: <%= ENV['REDIS_HOST'] || 'localhost' %>
+  port: <%= ENV['REDIS_PORT'] || 6379 %>
 production:
-  host: localhost
-  port: 6379
+  host: <%= ENV['REDIS_HOST'] || 'localhost' %>
```

Comment:
These changes are still not required. You can use `ENV['REDIS_URL']` exactly like the ENV['DATABASE_URL'] which will take precedence over this config file. No need to create two custom ENV vars for this.

Proof this ENV var override the config file here:  https://github.com/redis/redis-rb/blob/37ab9045e0fed202e42e49ccc11d41cc64de1759/lib/redis/client.rb#L400-L425